conf/flight_plans/rotorcraft_guided_flightplan.xml

```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE flight_plan SYSTEM "flight_plan.dtd">
+
+<flight_plan alt="1.0" ground_alt="0" lat0="51.990634" lon0="4.376789" max_dist_from_home="20" name="Rotorcraft Optitrack (Delft)" security_height="0.3">
+  <header>
+
+#include "autopilot.h"
+
+/********************************************//**
+ *  ABI messages for exception triggers or added values for the guided flight plan
+ ***********************************************/
+
+// ABI for obstacle detection (In here it comes from the module colorfilter)
+#include "subsystems/abi.h"
+
+#ifndef FP_GUIDED_OBSTACLE_DETECTION_ID
+#define FP_GUIDED_OBSTACLE_DETECTION_ID ABI_BROADCAST    
+PRINT_CONFIG_VAR(FP_GUIDED_OBSTACLE_DETECTION_ID)
+#endif
+
+static abi_event obstacle_detection_ev;
+static void obstacle_detection_cb(uint8_t sender_id, float distance_obstacle, float heading_obstacle);
+float distance_obstacle_FP;
+static void obstacle_detection_cb(uint8_t UNUSED(sender_id), float distance_obstacle, float UNUSED(heading_obstacle))
+{
+  distance_obstacle_FP = distance_obstacle;
+}
+
+</header>
+  <waypoints>
+    <waypoint name="HOME" x="0.0" y="0.0"/>
+    <waypoint name="STDBY" x="0.0" y="0.0"/>
+    <waypoint name="p1" x="2.0" y="2.5"/>
+    <waypoint name="p2" x="2.0" y="-2.5"/>
+    <waypoint name="p3" x="-2.0" y="-2.5"/>
+    <waypoint name="p4" x="-2.0" y="2.5"/>
+  </waypoints>
+  <sectors>
+    <sector color="red" name="Flight_Area">
+      <corner name="p1"/>
+      <corner name="p2"/>
+      <corner name="p3"/>
+      <corner name="p4"/>
+    </sector>
+  </sectors>
+  
+  <exceptions>
+    <!--uncomment when flying for real-->
+    <!--exception cond="datalink_time > 3" deroute="landed"/>
+    <exception cond="radio_control.status == RC_REALLY_LOST" deroute="kill"/-->
+  </exceptions>
+
+<!--Simple Guided flight plan:
+
+1. Flies in a small square, just based on guided flight commands (no waypoints)
+2. If it detects an obstacle (example shows only from color detector), it will turn the 90 degrees more quickly
+-->
+
+  <blocks>
+    <block name="Geo init">
+      <call_once fun="NavKillThrottle()"/>
+      <while cond="!GpsFixValid()"/>
+      <call_once fun="ResetAlt()"/>
+      <call_once fun="AbiBindMsgOBSTACLE_DETECTION(FP_GUIDED_OBSTACLE_DETECTION_ID, &obstacle_detection_ev, obstacle_detection_cb)"/>
+      <while cond="1"/>
+    </block>
+
+    <block name="Start Engine">
+      <call fun="StartEngines()"/>
+      <while cond="1"/>
+    </block>
+    
+    <block name="Take off">
+      <call fun="Climb(0.2)"/>
+      <call fun="WaitUntilAltitude(nominal_alt)"/>
```

Comment:
I still don't get why you use this, how often do you need to change the take off altitude without reflashing the controller?